src/codec_aom.c

```diff
@@ -97,13 +97,30 @@ static avifBool aomCodecGetNextImage(struct avifCodec * codec,
                                      avifImage * image)
 {
     if (!codec->internal->decoderInitialized) {
+        if (!sample) {
+            // Nothing to decode: no need to initialize anything.
+            return AVIF_FALSE;
+        }
+        aom_codec_iface_t * const decoderInterface = aom_codec_av1_dx();
+        struct aom_codec_stream_info streamInfo = { 0 };
+        if (aom_codec_peek_stream_info(decoderInterface, sample->data.data, sample->data.size, &streamInfo) != AOM_CODEC_OK) {
+            // The sample is incorrect or there are not enough bytes yet: no need to initialize anything.
+            return AVIF_FALSE;
+        }
+        if (streamInfo.w == 0 || streamInfo.h == 0) {
+            // The sequence header was not found: no need to initialize anything.
```

Comment:
Nit: Remove the ": no need to initialize anything" part of the comment, or change "anything" to "codec->internal->decoder" or "the libaom decoder".

We can also add "treat it as an error" after "The sequence header was not found".